### PR TITLE
fix: fixing lytics user_id and anonymousId mapping

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

Write a brief explainer on your code changes.

## What is the related Linear task?

Resolves INT-2669

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: src/cdk/v2/destinations/lytics/procWorkflow.yaml
Comment (krishna2020) [inline src/cdk/v2/destinations/lytics/procWorkflow.yaml:1]: clean up native js code

File: src/cdk/v2/destinations/lytics/procWorkflow.yaml [hunk 1]
```diff
@@ -28,7 +28,8 @@ steps:
         ...flattenTraits,
```
Comment (krishna2020) [inline src/cdk/v2/destinations/lytics/procWorkflow.yaml:28]: move out fields mapping to json config file

Comment (shrouti1507) [reply src/cdk/v2/destinations/lytics/procWorkflow.yaml:32]: done

File: src/cdk/v2/destinations/lytics/procWorkflow.yaml
Comment (shrouti1507) [reply src/cdk/v2/destinations/lytics/procWorkflow.yaml:1]: done

File: src/cdk/v2/destinations/lytics/data/LYTICSIdentifyConfig.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+[
+  {
+    "destKey": "user_id",
+    "sourceKeys": "userIdOnly",
+    "sourceFromGenericMap": true,
```
Comment (krishna2020) [inline src/cdk/v2/destinations/lytics/data/LYTICSIdentifyConfig.json:5]: will this not consider anonymousId ?

**Review by krishna2020 — APPROVED** [review]

File: test/integrations/destinations/lytics/processor/data.ts
Comment (krishna2020) [inline test/integrations/destinations/lytics/processor/data.ts:1]: There's scope for refactoring of test cases to make it more readable
- To move the common objects to variables & re-use them
- Limit the payload to required test case only
- Create test case files based on message type

**Review by sanpj2292 — APPROVED** [review]

**Pull request merged**